src/math/bspline.hpp

```diff
@@ -1,16 +1,44 @@
 #pragma once
-
 #include <Eigen/Core>
 
-namespace precice::math::bspline {
+namespace precice::math {
+
+class Bspline {
 
-/** Uses B-Spline interpolation to compute compute x(t) from given data (x0,t0), (x1,t1), ..., (xn,tn).
+public:
+  /**
+ * @brief Uses B-Spline interpolation to compute compute x(t) from given data (x0,t0), (x1,t1), ..., (xn,tn).
+ * Also updates the buffer if the flag _recomputeBuffer is true and t does not equal any of the timestamps in ts.
  *
- * @pre Requires at least 2 samples.
- * @pre Timestamps must be sorted from lowest to highest.
- * @pre splineDegree must be larger than 0.
- * @pre t must be within [ts.min; ts.max].
+ * @param ts the Timestamps which must be sorted from lowest to highest and contain at least 2 samples.
+ * @param xs the data to be interpolated. It has to contain at least two samples.
+ * @param splineDegree the used spline degree, which has to be larger than 0
+ * @param t must be within [ts.min; ts.max].
+ *
+ * @return the inteprolant x(t) if t does not equal any of the timestamps in ts.
+ * If t equals any of the timestamps in ts the corresponding value to t is returned.
  */
-Eigen::VectorXd interpolateAt(Eigen::VectorXd ts, const Eigen::MatrixXd &xs, int splineDegree, double t);
 
-} // namespace precice::math::bspline
+  Eigen::VectorXd interpolateAt(Eigen::VectorXd ts, const Eigen::MatrixXd &xs, int splineDegree, double t);
+
+  /**
+  * @brief Sets the flag that signals that the buffer should be recomputed next time interpolateAt is called.
+  */
+
+  void requireRecomputeBuffer();
+
+private:
+  /**
+   * @brief Calculates the knot vector and control points for the Bspline interpolation
+   *
+   * @param pts the values that are supposed to be interpolated
+   *
+   * @param knot_parameters the timestamps or values along which we want to interpolate
+   */
+  void computeMatrices(const Eigen::MatrixXd pts, int degree, Eigen::VectorXd knot_parameters, int ndofs);
+
+  bool            _recomputeBuffer = true; // Signals that the buffer should be recomputed next time interpolateAt is called
```

Comment:
Rather than calling this a "buffer" I would call it a "cache", because it is more specific. A buffer just stores some data, a cache stores data to speed up computations, but we might have to recompute, if there is a cache miss.